packages/adapter-node/src/handler.js

```diff
@@ -21,7 +21,8 @@ const protocol_header = env('PROTOCOL_HEADER', '').toLowerCase();
 const host_header = env('HOST_HEADER', 'host').toLowerCase();
 const body_size_limit = parseInt(env('BODY_SIZE_LIMIT', '524288'));
 
-const dir = fileURLToPath(import.meta.SERVER_DIR);
+const root = fileURLToPath(import.meta.SERVER_DIR);
+const dir = root.endsWith('.js') ? path.dirname(root) : root;
```

Comment:
I think a more robust approach would be to use `fs.statSync` to check if it's a directory instead of relying on a file extension (which may result in false positives/negatives)

Reply:
We could, yes. I added a simple extension check because so far all of the path errors tend to cause `dir` to point to a JS file instead of the actual directory. Will refactor to `fs.statSync`.